src/engine/source/api/src/test/handlers.cpp

```diff
@@ -458,6 +467,133 @@ api::Handler sessionsDelete(shared_ptr<Router> router, shared_ptr<Catalog> catal
     };
 }
 
+api::Handler runPost(std::shared_ptr<::router::Router> router)
+{
+    return [router](api::wpRequest wRequest) -> api::wpResponse
+    {
+        using RequestType = eTest::RunPost_Request;
+        using ResponseType = eTest::RunPost_Response;
+        const auto res = ::api::adapter::fromWazuhRequest<RequestType, ResponseType>(wRequest);
+
+        eTest::DebugMode debugMode;
+        uint32_t protocolQueue;
+        std::string protocolLocation;
+
+        // If the request is not valid, return the error
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        // Validate the params request
+        const auto& eRequest = std::get<RequestType>(res);
+        const auto errorMsg = !eRequest.has_name()    ? std::make_optional("Missing /session name")
+                        : !eRequest.has_event() ? std::make_optional("Missing /event")
+                                                : std::nullopt;
+
+        if (errorMsg.has_value())
+        {
+            return ::api::adapter::genericError<ResponseType>(errorMsg.value());
+        }
+
+        if (!eRequest.has_debugmode())
+        {
+            debugMode = eTest::DebugMode::ONLY_OUTPUT;
+        }
+        else
+        {
+            debugMode = eRequest.debugmode();
+        }
```

Comment:
```suggestion
        debugMode = eRequest.has_debugmode() ? eRequest.debugmode() : eTest::DebugMode::ONLY_OUTPUT;
```
I think this is more elegant, the same recommendation applies for the following if-elses